service/history/ndc/workflow_resetter.go

```diff
@@ -648,6 +656,11 @@ func (r *workflowResetterImpl) reapplyContinueAsNewWorkflowEvents(
 		// All subsequent events should be excluded from being re-applied. So, do nothing and return.
 		return lastVisitedRunID, nil
 	}
+
+	// track the child workflow initiated (but not yet completed) events after reset.
```

Comment:
What's the story around completed child workflows? 
I mean if workflow being reset is using workflowIDReusePolicyRejectDuplicate then it will fail?

Reply:
ah! yes. I should just record all initialized children after reset-point (whether they have reached a terminal state or not). Will make that change.

Reply:
Fixed it. Now tracking all children that were initialized.